code/modules/vehicles/speedbike.dm

```diff
@@ -34,4 +34,29 @@
 
 /obj/vehicle/space/speedbike/red
 	icon_state = "speedbike_red"
-	overlay_state = "cover_red"
\ No newline at end of file
+	overlay_state = "cover_red"
+
+//BM SPEEDWAGON
+
+/obj/vehicle/space/speedbike/speedwagon
+	name = "BM Speedwagon"
+	desc = "Push it to the limit, walk along the razor's edge."
+	icon = 'icons/obj/bike.dmi'
+	icon_state = "speedwagon"
+	layer = LYING_MOB_LAYER
+	overlay_state = "speedwagon_cover"
+
+/obj/vehicle/space/speedbike/speedwagon/Bump(mob/living/A)
+	. = ..()
+	if(A.density && has_buckled_mobs() && (istype(A, /mob/living/carbon/human) && has_buckled_mobs()))
+		var/atom/throw_target = get_edge_target_turf(A, pick(cardinal))
+		A.throw_at(throw_target, 4, 3)
+		A.Weaken(5)
+		A.adjustStaminaLoss(30)
+		A.apply_damage(rand(20,35), BRUTE)
+		visible_message("<span class='danger'>[src] crashes into [A]!</span>")
+		playsound(src, 'sound/effects/bang.ogg', 50, 1)
+
+/obj/vehicle/space/speedbike/speedwagon/buckle_mob(mob/living/M, force = 0, check_loc = 1)
+ 	. = ..()
+		riding_datum = new/datum/riding/space/speedbike/speedwagon
```

Comment:
while i get that you want drifting how is the piece of code you reviewed any relevant